### PR TITLE
Rename NameAliasType to TypeAliasType.

### PR DESCRIPTION
**Pull request by pschuh**
In swift, the syntax is typealias, but in the source code it is referred to as NameAliasType. Make these align by renaming to TypeAliasType.

### CODE REVIEW

File: lib/AST/ASTDumper.cpp [hunk 1]
```diff
@@ -3227,7 +3227,7 @@ namespace {
       PrintWithColorRAII(OS, ParenthesisColor) << ')';
     }
 
-    void visitNameAliasType(NameAliasType *T, StringRef label) {
+    void visitTypeAliasType(TypeAliasType *T, StringRef label) {
       printCommon(label, "name_alias_type");
```
Comment (slavapestov) [inline lib/AST/ASTDumper.cpp]: This should be changed too

**Review by slavapestov — APPROVED** [review]
Thank you!

**Comment by pschuh** [comment]
Thanks for the review; Everything should be updated, and ready for the CI to be called on both this and https://github.com/apple/swift-lldb/pull/1188.

**Comment by slavapestov** [comment]
https://github.com/apple/swift-lldb/pull/1188
@swift-ci Please smoke test

**Pull request merged**